blocks/algorithms/__init__.py

```diff
@@ -801,10 +801,10 @@ def __init__(self, learning_rate=0.002,
             add_role(param, ALGORITHM_HYPERPARAMETER)
 
     def compute_step(self, parameter, previous_step):
-        mean = shared_floatx_zeros_matching(parameter, 'mean')
-        add_role(mean, ALGORITHM_BUFFER)
-        variance = shared_floatx_zeros_matching(parameter, 'variance')
-        add_role(variance, ALGORITHM_BUFFER)
+        mean = _create_algorithm_buffer_for(parameter, 'mean')
+        mean.tag.for_parameter = parameter
```

Comment:
Good point, these predate my refactoring. I will fix them up later this evening (off to dinner).